694.py

```diff
@@ -0,0 +1,44 @@
+class Solution:
+    DIRS = {
+        (1, 0) : 'D',
+        (-1, 0): 'U',
+        (0, 1) : 'R',
+        (0, -1): 'L'
+    }
+
+    def numDistinctIslands(self, grid: List[List[int]]) -> int:
+        self.R, self.C = len(grid), len(grid[0])
+        distinct_islands = set()
+
+        for i in range(self.R):
+            for j in range(self.C):
+                if grid[i][j] == 1:
+                    island = self.bfs(grid, i, j)
+                    distinct_islands.add(island)
+
+        return len(distinct_islands)
+
+    def bfs(self, grid: List[List[int]], row: int, col: int):
+        queue = deque([(row, col)])
+        island = []
+
+        while queue:
+            r, c = queue.popleft()
+            grid[r][c] = 0
```

Comment:
There is no need to have this line to mark `grid[r][c]` to `0`; line 41 marks this cell already. This only time this line is useful is for the first element in `queue`. Therefore, you can move this line to line 23 right after your `queue` initialization.